src/main/java/at/hannibal2/skyhanni/data/BitsAPI.kt

```diff
@@ -0,0 +1,141 @@
+    private val fameRankGuiNamePattern by group.pattern("gui.famerankmenuname", "^(Community Shop|Booster Cookie)$")
+    private val fameRankGuiStackPattern by group.pattern("gui.famerankmenustack", "^(§aCommunity Shop|§eFame Rank)$")
+
+    @SubscribeEvent
+    fun onScoreboardChange(event: ScoreboardChangeEvent) {
+        for (line in event.newList) {
+            val message = line.trimWhiteSpaceAndResets()
+
+            bitsScoreboardPattern.matchMatcher(message) {
+                val amount = group("amount").formatNumber().toInt()
+                val earned = group("earned")?.formatNumber()?.toInt() ?: 0
+                bits = amount
+                if (earned > 0) bitsToClaim -= earned
+
+                return
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        val message = event.message.trimWhiteSpaceAndResets()
+
+        bitsFromFameRankUpChatPattern.matchMatcher(message) {
+            val amount = group("amount").formatNumber().toInt()
+            bitsToClaim += amount
+
+            return
+        }
+
+        bitsEarnedChatPattern.matchMatcher(message) {
+            // Only two locations where the bits line isn't shown, but you can still get bits
+            if (LorenzUtils.inAnyIsland(IslandType.CATACOMBS, IslandType.THE_RIFT)) return
+
+            val amount = group("amount").formatNumber().toInt()
+            bits += amount
+            bitsToClaim -= amount
+
+            return
+        }
+
+        boosterCookieAte.matchMatcher(message) {
+            bitsToClaim += (defaultcookiebits * currentFameRank.bitsMultiplier).toInt()
+
+            return
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryFullyLoaded(event: InventoryFullyOpenedEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+
+        val stacks = event.inventoryItems
+
+        if (bitsGuiNamePattern.matches(event.inventoryName)) {
+            val cookieStack = stacks.values.firstOrNull { bitsGuiStackPattern.matches(it.displayName) }
+            if (cookieStack != null) {
+                for (line in cookieStack.getLore()) {
+                    bitsAvailableMenu.matchMatcher(line) {
+                        bitsToClaim = group("toClaim").formatNumber().toInt()
+
+                        return
+                    }
+                }
+            }
+        }
+
+        if (fameRankGuiNamePattern.matches(event.inventoryName)) {
+            val fameRankStack = stacks.values.firstOrNull { fameRankGuiStackPattern.matches(it.displayName) }
+            if (fameRankStack != null) {
+                for (line in fameRankStack.getLore()) {
+                    fameRankCommunityShop.matchMatcher(line) {
+                        val rank = group("rank")
+                        currentFameRank = FameRank.entries.firstOrNull { it.rank == rank } ?: FameRank.NEW_PLAYER
+
+                        return
```

Comment:
this return is wrong it should continue@line (the for) because currently only one of the machtmakers will execute for the item